src/Perimeter.js

```diff
@@ -0,0 +1,174 @@
+// @flow
+import React, { Component, PropTypes } from 'react';
+
+// We listen to resize events too, but don't actually
+// use the event object so this is typed fine for now
+type EventListener = (MouseEvent) => void;
+
+type Props = {
+  onBreach: () => void,
+  once?: boolean,
+  padding: number,
+  mapListeners?: (EventListener) => EventListener,
+  children?: React$Element<*>
+};
+
+export default class Perimeter extends Component {
+  props: Props;
+  node: null | HTMLElement;
+  bounds: null | ClientRect;
+  initialOffset: number;
+  breached: boolean;
+  listening: boolean;
+  constructor(props: Props) {
+    const { mapListeners } = props;
+    super(props);
+    // The HTML element used as the perimeter center
+    this.node = null;
+    // The result of calling getBoundingClientRect on this.node
+    this.bounds = null;
+    // Initial pageYOffset, required since pages may be scrolled on load
+    // and getBoundingClientRect will take that into consideration when
+    // calculating the top offset
+    this.initialOffset = 0;
+    // Whether the mouse is within the perimeter
+    this.breached = false;
+    // If we're still listening for mousemove/resize events
+    this.listening = false;
+    // If a mapListeners function is provided, update the instance event
+    // handlers to be the returned mapped versions. Common use case would be
+    // debounced or throttled listeners.
+    if (mapListeners) {
+      this.handleMouseMove = mapListeners(this.handleMouseMove);
+      this.handleResize = mapListeners(this.handleResize);
+    }
+  }
+
+  static propTypes = {
+    onBreach: PropTypes.func.isRequired,
+    once: PropTypes.bool,
+    padding: PropTypes.number.isRequired,
+    mapListeners: PropTypes.func
+  };
+
+  static contextTypes = {
+    mouseMove: PropTypes.shape({
+      subscribe: PropTypes.func.isRequired,
+    })
+  };
+
+  /**
+   * When the component mounts we calculate the ClientRect
+   * for the target node and attach event listeners for:
+   *   - `mousemove` for checking perimeter breaches
+   *   - `resize` for recalculating ClientRect
+   */
+  componentDidMount() {
+    const { handleMouseMove, handleResize, node } = this;
+    if (node) {
+      this.bounds = node.getBoundingClientRect();
+      this.initialOffset = window.pageYOffset;
+      if(this.context && this.context.mouseMove && this.context.mouseMove.subscribe){
```

Comment:
We don't need to check for `this.context` as it will always exist. We can simplify this to:

```js
const { mouseMove } = this.context;
if (mouseMove && mouseMove.subscribe) {
  mouseMove.subscribe(handleMouseMove)
}
```

This is with the changes mentioned in [here](https://github.com/aweary/react-perimeter/pull/7/files#r106789178) in mind